src/main/java/net/dv8tion/jda/internal/managers/channel/ChannelManagerImpl.java

```diff
@@ -533,10 +565,57 @@ public M setInvitable(boolean invitable)
         return (M) this;
     }
 
+    public M setPinned(boolean pinned)
+    {
+        if (pinned)
+            flags.add(ChannelFlag.PINNED);
+        else
+            flags.remove(ChannelFlag.PINNED);
+        set |= PINNED;
+        return (M) this;
+    }
+
+    public M setRequireTag(boolean requireTag)
+    {
+        if (requireTag)
+            flags.add(ChannelFlag.REQUIRE_TAG);
+        else
+            flags.remove(ChannelFlag.REQUIRE_TAG);
+        set |= REQUIRE_TAG;
+        return (M) this;
+    }
+
+    public M setAvailableTags(List<? extends BaseForumTag> tags)
+    {
+        if (type != ChannelType.FORUM)
+            throw new IllegalStateException("Can only set available tags on forum channels.");
+        Checks.noneNull(tags, "Available Tags");
+        this.availableTags = new ArrayList<>(tags);
+        set |= AVAILABLE_TAGS;
+        return (M) this;
+    }
+
+    public M setAppliedTags(Collection<? extends ForumTagSnowflake> tags)
+    {
+        if (type != ChannelType.GUILD_PUBLIC_THREAD)
+            throw new IllegalStateException("Can only set applied tags on public thread channels.");
```

Comment:
Isn't this actually limited to only forum posts?
The check here is incomplete and the error message directs users in kinda the wrong direction.